be/src/exprs/array_functions.tpp

```diff
@@ -862,6 +862,68 @@ private:
     }
 };
 
+// all/any_match(lambda_func, array1, array2...)-> all/any_match(array_map(lambda_func, array1, array2...))
+// -> all/any_match(bool_array), result is bool type.
+// any_match: if there are true  matched, return true,  else if there are null, return null, otherwise, return false;
+// all_match: if there are false matched, return false, else if there are null, return null, otherwise, return true;
+template <bool isAny>
+class ArrayMatch {
+public:
+    static ColumnPtr process([[maybe_unused]] FunctionContext* ctx, const Columns& columns) {
+        return _array_match(columns);
+    }
+
+private:
+    static ColumnPtr _array_match(const Columns& columns) {
+        DCHECK(columns.size() == 1);
+        RETURN_IF_COLUMNS_ONLY_NULL(columns);
+        size_t chunk_size = columns[0]->size();
+        ColumnPtr bool_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
```

Comment:
We always unpack the const input column? have not do some optimizations for const column?

Reply:
yes, we can optimize it later if needed.